From acea9909f685f685107fc13f759f790cce23306b Mon Sep 17 00:00:00 2001
From: Alamy Liu <alamy.liu@gmail.com>
Date: Mon, 9 Nov 2020 11:07:41 -0800
Subject: [PATCH] Build: build against mbedtls v2.24

mbedtls_memcpy() and mbedtls_memset() are not available
in mbedtls library.
---
 mbedtls_MXCRYPTO/crypto_common.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/mbedtls_MXCRYPTO/crypto_common.h b/mbedtls_MXCRYPTO/crypto_common.h
index d797881..c9362fd 100644
--- a/mbedtls_MXCRYPTO/crypto_common.h
+++ b/mbedtls_MXCRYPTO/crypto_common.h
@@ -48,6 +48,16 @@
 #define mbedtls_malloc(...)  mbedtls_calloc(1, __VA_ARGS__)
 #endif
 
+#ifndef mbedtls_memcpy
+#include <string.h>
+#define mbedtls_memcpy       memcpy
+#endif
+
+#ifndef mbedtls_memset
+#include <string.h>
+#define mbedtls_memset       memset
+#endif
+
 #include "cy_crypto_core.h"
 
 #if defined(CY_USING_HAL) && !defined(CY_CRYPTO_HAL_DISABLE)
--
2.25.1
